kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/model/VirtualCluster.java

```diff
@@ -193,7 +194,12 @@ private Optional<SslContext> buildUpstreamSslContext() {
                         .orElse(SslContextBuilder.forClient());
                 var withTrust = Optional.ofNullable(targetClusterTls.trust()).map(NettyTrustProvider::new).map(tp -> tp.apply(sslContextBuilder))
                         .orElse(sslContextBuilder);
-                withTrust.endpointIdentificationAlgorithm("HTTPS");
+                if (tls.isPresent() && tls.get().trust() instanceof InsecureTls) {
```

Comment:
I don't think this is really the right option, but `TrustProvider` is a public interface in the config model and thus public API.